packages/facade/src/apis/__tests__/f-hooks.spec.ts

```diff
@@ -50,3 +50,86 @@ describe('Test FUniver', () => {
         onSteadyDisposable.dispose();
     });
 });
+
+describe('Test Unit Hooks', () => {
+    let get: Injector['get'];
+    let univerAPI: FUniver;
+    let univer: Univer;
+
+    beforeEach(() => {
+        const testBed = createUnitTestBed();
+        get = testBed.get;
+        univerAPI = testBed.univerAPI;
+        univer = testBed.univer;
+    });
+
+    it('Hooks Create Unit', () => {
+        const randomUnitId = 'randomUnitId';
+        let unitId;
+        let unitType;
+        univerAPI.getHooks().afterCreateUnit((unit) => {
+            unitId = unit.getUnitId();
+            unitType = unit.getType();
+        });
+        univer.createUnit(UniverInstanceType.UNIVER_SHEET, {
+            id: randomUnitId,
+        });
+        expect(unitId).toBe(randomUnitId);
+        expect(unitType).toBe(UniverInstanceType.UNIVER_SHEET);
+    });
+
+    it('Hooks Deny Create Unit', () => {
+        const disposeObj = univerAPI.getHooks().beforeCreateUnit(() => false);
+        try {
+            univer.createUnit(UniverInstanceType.UNIVER_SHEET, {
+                id: 'randomUnitId',
+            });
+        } catch (e) {
+            expect(e).toBeInstanceOf(Error);
+            expect((e as Error).message).toBe('[UniverInstanceService]: cannot add unit with user callback.');
+        }
+
+        disposeObj.dispose();
+
+        univer.createUnit(UniverInstanceType.UNIVER_SHEET, {
+            id: 'randomUnitId',
+        });
+    });
+
+    it('Hooks dispose Unit', () => {
+        const randomUnitId = 'randomUnitId';
+        let unitId;
+        let unitType;
+        const disposeCallback = vi.fn();
+        const beforeCallbackDispose = univerAPI.getHooks().beforeDisposeUnit((unit) => {
+            unitId = unit.getUnitId();
+            unitType = unit.getType();
+
+            return false;
+        });
+        univerAPI.getHooks().afterCreateUnit(disposeCallback);
+
+        const unit = univer.createUnit(UniverInstanceType.UNIVER_SHEET, {
+            id: randomUnitId,
+        });
+
+        // TODO: not calling dispose() method
+        // unit.dispose();
+
+        const _univerInstanceService = get(IUniverInstanceService);
```

Comment:
Not necessary to add an underscore before a variable. The underbar has a special usage for "not used variables".